### PR TITLE
introduce basic error handling

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :pushpin: Summary

Introduces very basic error handling for unhandled routes.

### :hammer_and_wrench: Detailed description
Screen Shot 2022-12-09 at 15 12 26

#### Open questions

- [x] I can't figure out how one would trigger the existing Dynamic Error Template logic in the `show` controller https://github.com/hashicorp/design-system/pull/814
- [x] ~~This route includes a number of error handling scenarios if the "fetch" from the json fails but I'm not sure how to trigger those or if they should be included here~~ they'll hit this page as well

### :link: External links

* [HDS-1160](https://hashicorp.atlassian.net/browse/HDS-1160)

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/app/templates/error.hbs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      alt="Animated gif created by Jory Tindall, showing a classic John Travolta meme on top of a composition of HDS component"
+    />
+    <div class="doc-page-error__content">
+      <h1 class="doc-text-h1">Well shoot...</h1>
+      <p class="doc-text-body">We can't seem to find what you're looking for or some unexpected error occurred. Head
+        back to the
+        <LinkTo class="doc-link-generic" @route="index">homepage</LinkTo>.</p>
+      <h4 class="doc-text-h4">Other helpful resources</h4>
+      <ul>
+        <li><LinkTo class="doc-link-generic" @route="about">About</LinkTo></li>
```
Comment (Dhaulagiri) [inline website/app/templates/error.hbs:17]: @didoo do we have any existing styles for styling lists like this?
Screen Shot 2022-12-14 at 11 23 37

Comment (didoo) [reply website/app/templates/error.hbs:17]: I intended to create a mixin for sharing this list style between the markdown list and the other normal HTML lists that use a specific class (eg. `doc-list-generic`) similar to what I have already done for the `doc-link-generic`. 
Probably I would suggest to leave it out for now (the styling) and create a JIRA ticket assigned to me, to add the style to it (should be a 5 mins job)

Comment (Dhaulagiri) [reply website/app/templates/error.hbs:17]: https://hashicorp.atlassian.net/browse/HDS-1248

**Review by didoo — APPROVED** [review]
🪩🕺💃

**Review by heatherlarsen — APPROVED** [review]
I just left a nit-picky wording suggestion. All good to go, though.

**Pull request merged**